lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,672 @@
+	 *     extract_directive_value( '{ "isOpen": false }', 'myPlugin' )              => array( 'myPlugin', array( 'isOpen' => false ) )
+	 *     extract_directive_value( 'otherPlugin::{ "isOpen": false }', 'myPlugin' ) => array( 'otherPlugin', array( 'isOpen' => false ) )
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|true $directive_value   The directive attribute value. It can be `true` when it's a boolean
+	 *                                       attribute.
+	 * @param string|null $default_namespace Optional. The default namespace if none is explicitly defined.
+	 * @return array An array containing the namespace in the first item and the JSON, the reference path, or null on the
+	 *               second item.
+	 */
+	private function extract_directive_value( $directive_value, $default_namespace = null ): array {
+		if ( empty( $directive_value ) || is_bool( $directive_value ) ) {
+			return array( $default_namespace, null );
+		}
+
+		$matches       = array();
+		$has_namespace = preg_match( '/^([\w\-_\/]+)::(.*)$/', isset( $directive_value ) ? $directive_value : '', $matches );
+
+		// Replaces the value and namespace if there is a namespace in the value.
+		if ( $has_namespace ) {
+			$default_namespace = $matches[1];
+			$directive_value   = isset( $matches[2] ) ? $matches[2] : null;
+		}
+
+		/*
+		 * Tries to decode the value as a JSON object. If it fails and the value
+		 * isn't `null`, it returns the value as it is. Otherwise, it returns the
+		 * decoded JSON or null for the string `null`.
+		 */
+		$decoded_json = json_decode( $directive_value, true );
+		if ( null !== $decoded_json || 'null' === $directive_value ) {
+				$directive_value = $decoded_json;
+		}
+
+		return array( $default_namespace, $directive_value );
+	}
+
+
+	/**
+	 * Processes the `data-wp-interactive` directive.
+	 *
+	 * It adds the default store namespace defined in the directive value to the
+	 * stack so it's available for the nested interactivity elements.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_interactive_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		// In closing tags, it removes the last namespace from the stack.
+		if ( $p->is_tag_closer() ) {
+			return array_pop( $namespace_stack );
+		}
+
+		// Tries to decode the `data-wp-interactive` attribute value.
+		$attribute_value = $p->get_attribute( 'data-wp-interactive' );
+		$decoded_json    = is_string( $attribute_value ) && ! empty( $attribute_value )
+		? json_decode( $attribute_value, true )
+		: null;
```

Comment:
nitpick, but these appear to be missing an indent level

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/58066/commits/246317af7c1730d95fd079511bcef8609026e460.